esp/esp/program/modules/handlers/lotteryfrontendmodule.py

```diff
@@ -0,0 +1,98 @@
+from esp.program.models import Program, ClassSection
+from esp.program.modules.base import ProgramModuleObj, needs_admin, main_call, aux_call
+from esp.program.controllers.lottery import LotteryAssignmentController, LotteryException
+from esp.utils.web import render_to_response
+from esp.users.models import ESPUser
+from esp.utils.decorators import json_response
+import numpy
+
+class LotteryFrontendModule(ProgramModuleObj):
+
+    @classmethod
+    def module_properties(cls):
+        return {
+            "admin_title": "Lottery Frontend",
+            "link_title": "Run the Lottery Assignment Thing",
+            "module_type": "manage",
+            "seq": 10
+            }
+
+    @main_call
+    @needs_admin
+    def lottery(self, request, tl, one, two, module, extra, prog):
+        #   Check that the lottery module is included
+        students = self.program.students()
+        if 'lotteried_students' not in students and 'twophase_star_students' not in students:
```

Comment:
I think you can just do `program.hasModule` or something, which is nice and cached and stuff.